chainercv/links/model/fpn/faster_rcnn.py

```diff
@@ -94,52 +115,106 @@ def forward(self, x):
         anchors = self.rpn.anchors(h.shape[2:] for h in hs)
         rois, roi_indices = self.rpn.decode(
             rpn_locs, rpn_confs, anchors, x.shape)
-        rois, roi_indices = self.head.distribute(rois, roi_indices)
-        head_locs, head_confs = self.head(hs, rois, roi_indices)
-        return rois, roi_indices, head_locs, head_confs
+        return hs, rois, roi_indices
 
     def predict(self, imgs):
-        """Detect objects from images.
+        """Conduct inference on the given images.
 
-        This method predicts objects for each image.
+        The value returned by this method is decided based on
+        the argument :obj:`return_values` of :meth:`__init__`.
+
+        Examples:
+
+            >>> from chainercv.links import FasterRCNNFPNResNet50
+            >>> model = FasterRCNNFPNResNet50(
+            ...     pretrained_model='coco',
+            ...     return_values=['rois', 'bboxes', 'labels', 'scores'])
+            >>> rois, bboxes, labels, scores = model.predict(imgs)
 
         Args:
-            imgs (iterable of numpy.ndarray): Arrays holding images.
-                All images are in CHW and RGB format
-                and the range of their value is :math:`[0, 255]`.
+            imgs (iterable of numpy.ndarray): Inputs.
 
         Returns:
-           tuple of lists:
-           This method returns a tuple of three lists,
-           :obj:`(bboxes, labels, scores)`.
-
-           * **bboxes**: A list of float arrays of shape :math:`(R, 4)`, \
-               where :math:`R` is the number of bounding boxes in a image. \
-               Each bounding box is organized by \
-               :math:`(y_{min}, x_{min}, y_{max}, x_{max})` \
-               in the second axis.
-           * **labels** : A list of integer arrays of shape :math:`(R,)`. \
-               Each value indicates the class of the bounding box. \
-               Values are in range :math:`[0, L - 1]`, where :math:`L` is the \
-               number of the foreground classes.
-           * **scores** : A list of float arrays of shape :math:`(R,)`. \
-               Each value indicates how confident the prediction is.
+            tuple of lists:
+            The table below shows the input and possible outputs.
+
+        .. csv-table::
+            :header: name, shape, dtype, format
+
+            :obj:`imgs`, ":math:`[(3, H, W)]`", :obj:`float32`, \
+            "RGB, :math:`[0, 255]`"
+            :obj:`rois`, ":math:`[(R', 4)]`", :obj:`float32`, \
```

Comment:
Can we add a line between inputs part and outputs part? (or just make two tables)

Reply:
OK